File: bridge.cpp
#include<bits/stdc++.h>
#include<string>
// using std::string;
// using namespace std;
class Implementation{
    public:
    virtual ~Implementation(){}
    virtual std::string OperationImplementation()const = 0;
};

class ConcreteImplementationA:public Implementation{
public:
std::string OperationImplementation()const override{
    return "ConcreteImplementationA";
}   
};
class ConcreteImplementationB:public Implementation{
    public:
std::string OperationImplementation()const override{
    return "ConcreteImplementationB";
}
};
class Abstraction{
    protected:
    Implementation*implementation_;
    public:
    Abstraction(Implementation*implementation){
        this->implementation_=implementation;
    }
    virtual ~Abstraction(){}
    virtual std::string Operation()const{
        return "Abstraction"+this->implementation_->OperationImplementation();
    }
};
class ExtendedAbstraction:public Abstraction{
    public:
    // ExtendedAbstraction(Implementation*implementation){
    //     Abstraction(implementation);
    // }
    ExtendedAbstraction(Implementation*implementation):Abstraction(implementation){}
    std::string Operation()const override{
        return "ExtendedAbstraction"+this->implementation_->OperationImplementation();
    }
};
void ClientCode(const Abstraction&abstraction){
    std::cout<<abstraction.Operation();
}
int main(){
    Implementation*implementation=new ConcreteImplementationA;
    Abstraction*abstraction=new Abstraction(implementation);
    ClientCode(*abstraction);

    std::cout<<std::endl;
    delete implementation;
    delete abstraction;

    implementation=new ConcreteImplementationB;
    abstraction=new ExtendedAbstraction(implementation);
    ClientCode(*abstraction);

    delete implementation;
    delete abstraction;
    return 0;
}

File: Bridge/bridge.py
from __future__ import annotations
from abc import ABC, abstractmethod

class Abstraction:
    def __init__(self,implemetnation:Implementation)->None:
        self.implemetnation=implemetnation
    def operation(self)->str:
        return (f"Abstraction : Base operation"
                f"{self.implemetnation.operation_implementation()}")

class ExtendedAbstraction(Abstraction):
    def operation(self)->str:
        return (f"ExtendedAbstraction : Extended operation"
                f"{self.implemetnation.operation_implementation()}")

class Implementation(ABC):
    @abstractmethod
    def operation_implementation(self)->str:
        pass
    
class ConcreteImplementationA(Implementation):
    def operation_implementation(self)->str:
        return "ConcreteImplementationA"

class ConcreteImplementationB(Implementation):
    def operation_implementation(self)->str:
        return "ConcreteImplementationB"

def client_code(abstraction:Abstraction)->None:
    print(abstraction.operation(),end="")

if __name__=="__main__":
    implemetnation=ConcreteImplementationA()
    abstraction=Abstraction(implemetnation)
    client_code(abstraction)

    print("\n")

    implemetnation=ConcreteImplementationB()
    abstraction=ExtendedAbstraction(implemetnation)
    client_code(abstraction)